Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata, Fabrizio Barroero, Laura Capuano, Nirvana Coppola,
+María Inés de Frutos Fernández, Sam van Gool, Silvain Rideau-Kikuchi, Amos Turchet,
+Francesco Veneziano
+-/
+
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+import Mathlib.Analysis.Normed.Ring.Seminorm
+
+/-!
+# Ostrowski’s Theorem
+
+The goal of this file is to prove Ostrowski’s Theorem which gives a list of all the nontrivial
+absolute values on a number field up to equivalence.
+
+## References
+* https://kconrad.math.uconn.edu/blurbs/gradnumthy/ostrowskiQ.pdf
+* https://kconrad.math.uconn.edu/blurbs/gradnumthy/ostrowskinumbfield.pdf
+
+## Tags
+ring_norm, ostrowski
+-/
+
+noncomputable section
+namespace Rat.MulRingNorm
+open Int
+
+variable {f g : MulRingNorm ℚ}
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the
+integers. -/
+lemma NormRat_eq_on_Int_iff_eq_on_Nat : (∀ n : ℕ , f n = g n) ↔ (∀ n : ℤ , f n = g n) := by
+  refine' ⟨_, fun a n => a n⟩
+  intro h z
+  obtain ⟨n,rfl | rfl⟩ := eq_nat_or_neg z
+  · exact h n
+  · simp only [Int.cast_neg, Int.cast_natCast, map_neg_eq_map]
+    exact h n
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the natural
+numbers. -/
+lemma NormRat_eq_iff_eq_on_Nat : (∀ n : ℕ , f n = g n) ↔ f = g := by
```

Comment:
The name of the lemma is somewhat opposite to the statement (which says "equivalent on Nat iff equal"). The same comment applies to the lemma above.